### PR TITLE
[Merged by Bors] - feat: port Control.Traversable.Instances

### PR DESCRIPTION
**Pull request by zeramorphic**

### CODE REVIEW

**Comment by zeramorphic** [comment]
I removed some `@[nolint unused_arguments]` attributes, which were used because inclusion of typeclass variables was less powerful in Lean 3. Is `show LawfulMonad _ from inferInstance` an acceptable spelling, or is there something better (that doesn't involve renaming instances)?

**Comment by zeramorphic** [comment]
I've noticed that although the original docstring says it provides `Traversable` instances, those are already in a previous file, and we only provide `IsLawfulTraversable` instance here. Does that mean that the file deserves to be moved elsewhere?

**Comment by ChrisHughes24** [comment]
bors merge

**Comment by zeramorphic** [comment]
I think there's a functor_norm regression, `simp [functor_norm]` now doesn't close three of the goals in this file.

**Comment by Ruben-VandeVelde** [comment]
Let's not hold up this PR any longer - I added the explicit theorems with a porting note

File: Mathlib/Control/Traversable/Instances.lean [hunk 1]
```diff
@@ -74,35 +77,36 @@ variable [LawfulApplicative F] [LawfulApplicative G]
...
 
 variable (η : ApplicativeTransformation F G)
 
 protected theorem naturality {α β} (f : α → F β) (x : List α) :
-    η (List.traverse f x) = List.traverse (@η _ ∘ f) x := by induction x <;> simp! [*, functor_norm]
-#align list.naturality List.naturality
+    η (List.traverse f x) = List.traverse (@η _ ∘ f) x := by
+  -- Porting note: added `ApplicativeTransformation` theorems
+  induction x <;> simp! [*, functor_norm, ApplicativeTransformation.preserves_map,
+    ApplicativeTransformation.preserves_seq, ApplicativeTransformation.preserves_pure]
```
Comment (jcommelin) [inline Mathlib/Control/Traversable/Instances.lean:101]:
```suggestion
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**